### DISCUSSION TITLE
Custom shader support?

### DISCUSSION THREAD

**Discussion by phisch** [rust-skia/rust-skia#582] [Q&A] [answered]
I'm pretty inexperienced with skia, rust and rust-skia, but I am wondering if there is a way (or will be a way) to use custom shaders for paint and filters. 

Like could you use a kirby shader as a paint to fill in a shape with? https://www.shadertoy.com/view/lt2fD3
Or could you apply a wobble shader (that takes mouse input and time) onto something?

**Comment by pragmatrix** [comment, marked as answer]
There is a fairly new Skia type named [`SkRuntimeEffect`](https://github.com/google/skia/blob/fa2edac74af81de2d50aee934903dc5d081d28fa/include/effects/SkRuntimeEffect.h) which seems to provide exactly that. You can use Skia's shading language and pass them to the functions prefixed with `make_for` in the [`runtime_effect.rs`](https://github.com/rust-skia/rust-skia/blob/9d2efc395929fd00d088c09eec763e542bcaf1d1/skia-safe/src/effects/runtime_effect.rs) bindings.

Further references:

- Skia Shading Language: https://skia.org/docs/user/sksl/
- Playground: https://shaders.skia.org/
